### PR TITLE
Remove the usage of ManagedMemory from the OpticalFlow tests

### PR DESCRIPTION
**Pull request by JanuszL**
Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

## Description
- [x] **Bug fix** (*non-breaking change which fixes an issue*)
- [ ] **New feature** (*non-breaking change which adds functionality*)
- [ ] **Breaking change** (*fix or feature that would cause existing functionality to not work as expected*)
- [ ] **Refactoring** (*Redesign of existing code that doesn't affect functionality*)
- [ ] **Other** (*e.g. Documentation, Tests, Configuration*)

### What happened in this PR
- removed the usage of ManagedMemory from the OpticalFlow tests

#### Additional information
- Affected modules and functionalities:
  - optical_flow_turing_test.cc 

- Key points relevant for the review:
  - NA

### Tests
- [x] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: dali/operators/sequence/optical_flow/turing_of/optical_flow_turing_test.cc [hunk 1]
```diff
@@ -139,22 +139,20 @@ TEST_F(OpticalFlowTuringKernelTest, FlowVectorTest) {
...
+  tested.resize(reference_data.size());
+  tested_host.resize(reference_data.size());
+
+  optical_flow::kernel::EncodeFlowComponents(input.data(), tested.data(), pitch, width, height, 0);
   CUDA_CALL(cudaDeviceSynchronize());
+  copyD2H(tested_host.data(), tested.data(), tested_host.size());
 
   for (size_t i = 0; i < reference_data.size(); i++) {
-    EXPECT_EQ(reference_data[i], tested[i]) << "Failed on index: " << i;
+    EXPECT_EQ(reference_data[i], tested_host[i]) << "Failed on index: " << i;
```
Comment (mzient) [inline dali/operators/sequence/optical_flow/turing_of/optical_flow_turing_test.cc:154]: This condition reads uninitialized memory (the padding between width and pitch) - either loop over width/height and check only the relevant values, or use cudaMemset to fill the array.

Comment (JanuszL) [reply dali/operators/sequence/optical_flow/turing_of/optical_flow_turing_test.cc]: Done

File: dali/operators/sequence/optical_flow/turing_of/optical_flow_turing_test.cc [hunk 2]
```diff
@@ -139,22 +139,23 @@ TEST_F(OpticalFlowTuringKernelTest, FlowVectorTest) {
...
-
-  optical_flow::kernel::EncodeFlowComponents(input, tested, pitch, width, height, 0);
+  DeviceBuffer<float> input;
+  DeviceBuffer<int16_t> tested;
+  std::vector<int16_t> tested_host;
+  input.from_host(test_data);
+  tested.resize(reference_data.size());
+  tested_host.resize(reference_data.size());
+
+  cudaMemset(tested.data(), 0xCC, reference_data.size());
```
Comment (mzient) [inline dali/operators/sequence/optical_flow/turing_of/optical_flow_turing_test.cc:149]:
```suggestion
  cudaMemset(tested.data(), 0x5C, reference_data.size());
```

File: dali/operators/sequence/optical_flow/turing_of/optical_flow_turing_test.cc [hunk 3]
```diff
@@ -139,22 +139,21 @@ TEST_F(OpticalFlowTuringKernelTest, FlowVectorTest) {
...
 
-  for (size_t i = 0; i < reference_data.size(); i++) {
-    EXPECT_EQ(reference_data[i], tested[i]) << "Failed on index: " << i;
-  }
-
-  CUDA_CALL(cudaFree(tested));
-  CUDA_CALL(cudaFree(input));
+    for (size_t i = 0; i < reference_data.size(); i++) {
+      EXPECT_EQ(reference_data[i], tested_host[i]) << "Failed on index: " << i;
+    }
```
Comment (mzient) [inline dali/operators/sequence/optical_flow/turing_of/optical_flow_turing_test.cc:156]: Weird indent.

File: dali/operators/sequence/optical_flow/turing_of/optical_flow_turing_test.cc [hunk 4]
```diff
@@ -45,24 +46,23 @@ class OpticalFlowTuringKernelTest : public ::testing::Test {
...
+    input.from_host(input_data);
+    tested.resize(reference_data.size());
+    tested_host.resize(reference_data.size());
 
     auto w = Grayscale ? width_gray_ : width_;
     auto h = Grayscale ? height_gray_ : height_;
     auto p = Grayscale ? pitch_gray_ : pitch_;
 
-    cc(input, tested, p, w, h, 0);
+    cc(input.data(), tested.data(), p, w, h, 0);
```
Comment (mzient) [inline dali/operators/sequence/optical_flow/turing_of/optical_flow_turing_test.cc:59]: Same here - the output array is not initialized and the pitch is larger than width.

Comment (JanuszL) [reply dali/operators/sequence/optical_flow/turing_of/optical_flow_turing_test.cc:60]: Done

**Review by mzient — APPROVED** [review]

File: dali/operators/sequence/optical_flow/turing_of/optical_flow_turing_test.cc [hunk 5]
```diff
@@ -45,34 +46,34 @@ class OpticalFlowTuringKernelTest : public ::testing::Test {
...
+    input.from_host(input_data);
+    tested.resize(reference_data.size());
+    tested_host.resize(reference_data.size());
 
     auto w = Grayscale ? width_gray_ : width_;
     auto h = Grayscale ? height_gray_ : height_;
     auto p = Grayscale ? pitch_gray_ : pitch_;
 
-    cc(input, tested, p, w, h, 0);
+    cudaMemset(tested.data(), 0x5C, tested.size());
```
Comment (mzient) [inline dali/operators/sequence/optical_flow/turing_of/optical_flow_turing_test.cc:59]:
```suggestion
    cudaMemset(tested.data(), 0x5C, tested.size()*sizeof(tested[0]));
```

File: dali/operators/sequence/optical_flow/turing_of/optical_flow_turing_test.cc [hunk 6]
```diff
@@ -139,22 +140,21 @@ TEST_F(OpticalFlowTuringKernelTest, FlowVectorTest) {
...
-
-  optical_flow::kernel::EncodeFlowComponents(input, tested, pitch, width, height, 0);
+  DeviceBuffer<float> input;
+  DeviceBuffer<int16_t> tested;
+  std::vector<int16_t> tested_host;
+  input.from_host(test_data);
+  tested.resize(reference_data.size());
+  tested_host.resize(reference_data.size());
+
+  cudaMemset(tested.data(), 0x5C, tested.size()*sizeof(int16_t));
```
Comment (mzient) [inline dali/operators/sequence/optical_flow/turing_of/optical_flow_turing_test.cc:150]:
```suggestion
  cudaMemset(tested.data(), 0x5C, tested.size()*sizeof(tested[0]));
```

**Review by mzient — APPROVED** [review]

File: dali/operators/sequence/optical_flow/turing_of/optical_flow_turing_test.cc (hunk 6 above)
Comment (JanuszL) [reply dali/operators/sequence/optical_flow/turing_of/optical_flow_turing_test.cc]: Done

File: dali/operators/sequence/optical_flow/turing_of/optical_flow_turing_test.cc (hunk 5 above)
Comment (JanuszL) [reply dali/operators/sequence/optical_flow/turing_of/optical_flow_turing_test.cc]: Done

**Review by klecki — APPROVED** [review]

**Pull request merged**